awx/api/views.py

```diff
@@ -184,6 +185,18 @@ def unattach(self, request, *args, **kwargs):
         return response
 
 
+class RelatedJobsPreventDeleteMixin(object):
+    def delete(self, request, *args, **kwargs):
+        obj = self.get_object()
```

Comment:
why not override `perform_destroy(instance)` so that you don't have to do `get_object` redundantly?

/nit